File: app/src/main/java/zone/com/zrefreshlayoutdemo/WaveHeaderActivity.java
package zone.com.zrefreshlayoutdemo;

import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.widget.ImageView;
import android.widget.TextView;
import butterknife.BindView;
import butterknife.ButterKnife;
import zone.com.zrefreshlayout.ZRefreshLayout;
import zone.com.zrefreshlayoutdemo.header.WaveHead;

/**
 * Created by fuzhipeng on 2017/1/11.
 */

public class WaveHeaderActivity extends AppCompatActivity {
    @BindView(R.id.iv)
    ImageView iv;
    @BindView(R.id.refresh)
    ZRefreshLayout refresh;
    @BindView(R.id.tv)
    TextView tv;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.unique_feature);
        ButterKnife.bind(this);

        WaveHead waveHead = new WaveHead();
        waveHead.setResourceId(R.drawable.wave);
        refresh.setIHeaderView(waveHead);

        tv.setText("写着玩的没啥用~");
        refresh.setPullListener(new ZRefreshLayout.PullListener() {
            @Override
            public void refresh(final ZRefreshLayout zRefreshLayout) {
                refresh.postDelayed(new Runnable() {
                    @Override
                    public void run() {
                        iv.setImageResource(R.drawable.aaaaaaaaaaaab);
                        zRefreshLayout.refreshComplete();
                    }
                }, 2000);
            }

        });
    }
}

File: app/src/main/java/zone/com/zrefreshlayoutdemo/WaveHeaderActivity.kt
package zone.com.zrefreshlayoutdemo

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import kotlinx.android.synthetic.main.unique_feature.*
import zone.com.zrefreshlayout.ZRefreshLayout
import zone.com.zrefreshlayoutdemo.header.WaveHead

/**
 * Created by fuzhipeng on 2017/1/11.
 */

class WaveHeaderActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.unique_feature)
        val waveHead = WaveHead()
        waveHead.resourceId = R.drawable.wave
        refresh.iHeaderView = waveHead

        tv.text = "写着玩的没啥用~"
        refresh.pullListener = ZRefreshLayout.PullListener { zRefreshLayout ->
            refresh.postDelayed({
                iv.setImageResource(R.drawable.aaaaaaaaaaaab)
                zRefreshLayout.refreshComplete()
            }, 2000)
        }
    }
}
